### PR TITLE
Andrewl cache sgpr

### PR DESCRIPTION
**Pull request by Andrew878**
Hi @johnamcleod 

I have been putting work into the posterior implementation of SGPR, but I am going on holiday soon, so Alessandro is going to take over. I have written the tests, and have implemented where:
- posterior fused predictions == SGPR depreciated predictions
- posterior cached predictions  'almost' equals SGPR depreciated predictions (the mean function is equivalent, but the variance isn't quite finished)

Hi @avullo
Here is the draft PR!

**PR type:** bugfix / enhancement / new feature / doc improvement

**Related issue(s)/PRs:** 

## Summary

**Proposed changes**

* ...
* ...
* ...

**What alternatives have you considered?**

### Minimal working example

```python
# Put your example code in here
```

### Release notes

**Fully backwards compatible:** yes / no

**If not, why is it worth breaking backwards compatibility:**

## PR checklist

- [x] New features: code is well-documented
  - [ ] detailed docstrings (API documentation)
  - [x] notebook examples (usage demonstration)
- [x] The bug case / new feature is covered by unit tests
- [x] Code has type annotations
- [ ] Build checks
  - [ ] I ran the black+isort formatter (`make format`)
  - [ ] I locally tested that the tests pass (`make check-all`)
- [x] Release management
  - [x] RELEASE.md updated with entry for this change
  - [x] New contributors: I've added myself to CONTRIBUTORS.md

### CODE REVIEW

File: gpflow/posteriors.py [hunk 1]
```diff
@@ -292,6 +293,118 @@ def _conditional_fused(
...
+        """
+        Computes predictive mean and (co)variance at Xnew, *excluding* mean_function.
+        Relies on cached alpha and Qinv.
+        """
+        Kus = Kuf(self.inducing_variable, self.kernel, Xnew)
+        Knn = self.kernel(Xnew, full_cov=full_cov)
+
+        mean = tf.transpose(Kus) @ self.alpha
+        var = Knn - tf.transpose(Kus) @ self.Qinv @ Kus
+
```
Comment (Andrew878) [inline gpflow/posteriors.py:366]: This is not working. It is also needs to be implemented for ` full_cov: bool = False, full_output_cov: bool = False`

Comment (avullo) [reply gpflow/posteriors.py]: @Andrew878, @johnamcleod  I have made the necessary changes to make this work:
- there was an error in that the mean was returned with the mean function values added
- the provided formula for Qinv is incorrect, I have changed the fast predictions notebook accordingly
- I have taken into account the two `full_cov` cases

The tests pass for all `full_cov` cases, but the notebook is not happy in the `full_cov=False` case.

Comment (avullo) [reply gpflow/posteriors.py]: @Andrew878 @johnamcleod This is not working in both cases. Note how we don't need to factorise `Qinv` in the `full_cov=False` case any more.

**Review by johnamcleod — APPROVED** [review]
LGTM!

**Pull request merged**